spec/lib/msf/core/exploit/remote/remote/http/http_cookie_jar_spec.rb

```diff
@@ -0,0 +1,278 @@
+  def random_string(min_len = 1, max_len = 12)
+    str = Faker::Alphanumeric.alpha(number: max_len)
+    str[0, rand(min_len..max_len)]
+  end
+
+  def cookie
+    Msf::Exploit::Remote::HTTP::HttpCookie.new(
+      random_string,
+      random_string,
+      max_age: rand(1..100),
+      path: '/' + random_string,
+      secure: rand(0..1) == 1,
+      httponly: rand(0..1) == 1,
+      domain: random_string
+    )
+  end
+
+  let(:cookie_jar) { described_class.new }
+
+  before(:each) do
+    Timecop.freeze(Time.local(2008, 9, 5, 10, 5, 30))
+  end
+
+  after(:each) do
+    Timecop.return
+  end
+
+  describe 'empty?' do
+    it 'will return true when no cookies are in a cookie_jar' do
+      # cookie_jar made in before
+
+      e = cookie_jar.empty?
+
+      expect(e).to eq(true)
+    end
+
+    it 'will return false when a cookie has been added to a cookie_jar' do
+      c = cookie
+
+      cookie_jar.add(c)
+      e = cookie_jar.empty?
+
+      expect(e).to eq(false)
+    end
+  end
+
+  describe 'clear' do
+    it 'will make no changes to an empty cookiejar' do
+      # empty cookie_jar made in before
+
+      cookie_jar.clear
+
+      expect(cookie_jar.empty?).to eq(true)
+    end
+
+    it 'will return false when a cookie has been added to a cookie_jar' do
+      c = cookie
+
+      cookie_jar.add(c)
+      cookie_jar.clear
+
+      expect(cookie_jar.empty?).to eq(true)
+    end
+  end
+
+  describe 'cookies' do
+    it 'will return an empty array when no cookies have been added to the jar' do
+      # cookie_jar made in before
+
+      c_array = cookie_jar.cookies
+
+      expect(c_array.class).to eq(Array)
+      expect(c_array.empty?).to eq(true)
+    end
+
+    it 'will return an array of all cookies added to the cookie_jar when called with no url param' do
+      c_array = []
+      rand(1..10).times do
+        c = cookie
+
+        c_array.append(c)
+        cookie_jar.add(c)
+      end
+
+      jar_array = cookie_jar.cookies
+
+      expect(c_array.sort).to eq(jar_array.sort)
+    end
+
+    # it 'will return an array of all cookies that have parent paths to the url passed' do
+    #   c_array = [cookie, cookie, cookie, cookie]
+    #   c_array[0].secure = false
+    #   c_array[1].secure = false
+    #   c_array[1].domain = c_array[0].domain
+    #   c_array[1].path = '/'
+    #   c_array.map { |c| cookie_jar.add(c) }
+    #
+    #   require 'pry'
+    #   binding.pry
+    #
+    #   filtered_cookies = cookie_jar.cookies("http://#{c_array[0].domain}#{c_array[0].path}")
+    #
+    #   expect(filtered_cookies.length).to eq(2)
+    #   expect(filtered_cookies.include?(c_array[0])).to eq(true)
+    #   expect(filtered_cookies.include?(c_array[1])).to eq(true)
+    #   expect(filtered_cookies.include?(c_array[2])).to eq(false)
+    #   expect(filtered_cookies.include?(c_array[3])).to eq(false)
+    # end
+  end
+
+  describe 'add' do
+    it 'unacceptable cookie missing path throws ArgumentError' do
```

Comment:
I don't think this reads very well, in the test run output does it make more sense in context?